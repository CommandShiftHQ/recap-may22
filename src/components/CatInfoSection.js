import React from 'react'
import CatTemperamentForm from './CatTemperamentForm'

const CatInfoSection = ({ catData }) => {
    const { name, description, temperament, image } = catData

    return (
        <div>
            <p><b>Name:</b> {name}</p>
            <p><b>Description:</b> {description}</p>
            <CatTemperamentForm temperament={temperament} />
            <img src={image.url} alt="cat" width="60%" height="auto" />
        </div>
    )
}

export default CatInfoSection