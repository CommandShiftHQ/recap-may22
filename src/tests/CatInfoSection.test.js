import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import CatInfoSection from "../components/CatInfoSection"

describe("CatInfoSection", () => {
    const catData = {
        name: "Bengal",
        temperament: "Alert, Agile, Energetic, Demanding, Intelligent",
        description: "Bengals are a lot of fun to live with, but they're definitely not the cat for everyone, or for first-time cat owners. Extremely intelligent, curious and active, they demand a lot of interaction and woe betide the owner who doesn't provide it.",
        image: {
            "id": "O3btzLlsO",
            "width": 1100,
            "height": 739,
            "url": "https://cdn2.thecatapi.com/images/O3btzLlsO.png"
        }
    }

    it("contains the cat's name", () => {
        render(<CatInfoSection catData={catData} />);
        expect(screen.getByText(/name/i)).toBeInTheDocument();
    });

    it("displays cat information", () => {
        const { container } = render(<CatInfoSection catData={catData} />);
        expect(container).toMatchSnapshot();
    });
})

