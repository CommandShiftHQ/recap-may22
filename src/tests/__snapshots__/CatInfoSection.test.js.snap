// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatInfoSection displays cat information 1`] = `
<div>
  <div>
    <p>
      <b>
        Name:
      </b>
       
      Bengal
    </p>
    <p>
      <b>
        Description:
      </b>
       
      Bengals are a lot of fun to live with, but they're definitely not the cat for everyone, or for first-time cat owners. Extremely intelligent, curious and active, they demand a lot of interaction and woe betide the owner who doesn't provide it.
    </p>
    <form>
      <p>
        Would you agree that ... cats are: 
      </p>
      <label
        data-testid="temperament-checkbox"
      >
        <input
          id="Alert-0"
          type="checkbox"
          value="Alert"
        />
        Alert
      </label>
      <label
        data-testid="temperament-checkbox"
      >
        <input
          id=" Agile-1"
          type="checkbox"
          value=" Agile"
        />
         Agile
      </label>
      <label
        data-testid="temperament-checkbox"
      >
        <input
          id=" Energetic-2"
          type="checkbox"
          value=" Energetic"
        />
         Energetic
      </label>
      <label
        data-testid="temperament-checkbox"
      >
        <input
          id=" Demanding-3"
          type="checkbox"
          value=" Demanding"
        />
         Demanding
      </label>
      <label
        data-testid="temperament-checkbox"
      >
        <input
          id=" Intelligent-4"
          type="checkbox"
          value=" Intelligent"
        />
         Intelligent
      </label>
      <button>
        Submit
      </button>
    </form>
    <img
      alt="cat"
      height="auto"
      src="https://cdn2.thecatapi.com/images/O3btzLlsO.png"
      width="60%"
    />
  </div>
</div>
`;
